lambda/matching.ml

```diff
@@ -3183,11 +3183,12 @@ let combine_constructor value_kind loc arg pat_env cstr partial ctx def
                       (Lprim (Pintcomp Ceq, [ Lvar tag; ext ], loc), act, rem, value_kind))
                   nonconsts default
               in
-              (* Since the tag is used directly in the tests, it would be sound
-                 to mark it Alias+Reads_agree but this would not change performance.
-                 We keep it this way to guarantee soundness of in-place overwriting. *)
+              (* CR uniqueness: This read might be on a unique allocation that could
+                 be overwritten. We should ensure that it is never pushed down.
+                 However, we can not use [Reads_vary] here, since this breaks
+                 flambda2. *)
```

Comment:
We should change this comment to discuss performance rather than the flambda2 issue